src/msw/msgdlg.cpp

```diff
@@ -542,12 +542,8 @@ int wxMessageDialog::ShowModal()
 
     wxWindowDisabler disableOthers(this, GetParentForModalDialog());
 
-#ifdef wxHAS_MSW_TASKDIALOG
-    if ( HasNativeTaskDialog() )
+    if ( TaskDialogIndirect_t taskDialogIndirect = GetTaskDialogIndirectFunc() )
```

Comment:
I thought declaring variable in the if statement is available only since C++17?

Reply:
No, it is available since C++98, probably even pre-standard versions. The new C++17 thing is declaring a variable without the declaration itself being the condition. Like this: `if ( int val; getVal(&val) ) ...`

Reply:
I see, thanks for the explanation and sorry about the noise.